### ISSUE TITLE
Revisit email enforcement UX

### ISSUE THREAD

**Issue by zlavergne** [hotosm/tasking-manager#1791]
I've gotten feedback about the new enforcement being confusing and difficult. What seems to be the most common case is:
* users click log in
* authorize with OSM
* come back to the TM and go to  a project
* try to start mapping only to be told they can't without a clear direction for how to fix it

Since it sounds like the email requirement needs to stay, we need to improve the UX of how to enforce it. 

In the original issue #1365, there were a [few options](https://github.com/hotosm/tasking-manager/issues/1365#issuecomment-476526378) summarized by @xamanu. I'm leaning towards something like option 4 but instead of a page, it could be a modal. Since we functionally prohibit any use of the TM without an email, I think it's actually easier for the user to be forced to verify. It's more of a burden to have to navigate back and forth between a project and a user's profile to be able to start on a project.

**Comment by xamanu** [comment]
Gathering some more feedback:

> 'to receive notifications about your mapping'. well 1) what kind of notifications 2) what else may it be used for? 3) what if I don't want to receive notifications?

Trying to answer:
1) what kind of notifications
• Feedback from tasks and projects a user has been working on (comments people actually expect to reach the other person)
• Eventually suggestions for recommended projects (users will be able to opt-out)
2) what else may it be used for?
• Nothing. We are also producing a statement around that (GDPR compliant)
3) what if I don't want to receive notifications?
Some messages we consider fundamental for the well function of the system. All other messages shall be configurable for you to opt-out.

>  think if you can make clear why you're collecting emails, how it's being used and how you can delete your account, that'd be an improvement.

Yes, let's tackle this in this issue.

**Comment by xamanu** [comment]
Some proposal on how to improve this:

**1. Onboarding issue**
as described by @zlavergne in the issue post: For the redesign we are actually changing the behaviour to collect the email address already before the redirect to OSM. This will allow us to send the user an instruction mail, so they also find their way back to the Tasking Manager, which can be hard to remember, especially for first time mappers after creating the account on OSM. For existing users we could post a similar modal, exactly as @zlavergne suggests.

Please have a look at [this slide](https://projects.invisionapp.com/share/WTRWOQ6Y6RQ#/screens/367084748) to get a better idea of the new onboarding process (-> `Submit & proceed` sends user over to OSM for login or sign up).

**2. Immediate step**
We want to fix this for the Tasking Manager 3 (currently live). For this I suggest to change the message, to be less looking like an error, explaining better why and what we collect, and refer to the privacy policy again:

![image](https://user-images.githubusercontent.com/97706/62936564-ae288000-bdb9-11e9-979a-8b6012b385ba.png)

**Comment by xamanu** [comment]
Created a hotfix release containing changes for a more explanatory message and reference to the privacy policy of HOT: https://github.com/hotosm/tasking-manager/pull/1827
Further unverified email addresses are now enough. This allows people to get started without the hassle to wait for a confirmation mail, check their in-box, etc.

**Comment by xivk** [comment]
I'm not sure how to go about reporting this so I'm posting it here.

This email requirement thing is not going very well in the London missing maps group. The main issues are with the signup process, and the requirements to give your email twice for a new user signing up. 

Is this an absolute requirement to give up you email? Wouldn't it be possible to just encourage users to give their email or have an opt-out? Most people understand the need for email address and notifications coming directly from the tasking manager but it would be nice to have an opt-out.

**Comment by rachblevine** [comment]
Hey Team,
A couple other suggestions after having this really derail people for the last bunch of events we've organized. 1) Could you possible make it open in a new tab / do a pop up window? People are being taken away from their exact task and it's adding to the confusion. 2) I don't think confirmation is necessary. Either you trust them or you don't, but making them wait for another confirmation email is delaying events and really putting people off right at the start of their engagement with this tool and HOT. 3) Don't put the notification in a secondary tab (it's currently under the map tab). That means a new volunteer needs to start the mapping process to find this new requirement and pause. This make it complicated to do ahead of time.  4) As it currently stands, the TM main page & individual task pages don't let you know anything else is needed.  So a new volunteer thinks they are successfully logged in and ready to go, but surprise, they aren't and now the whole groups needs to pause and wait for confirmation emails.  So can you do a notification or something during that initial log in perhaps?
Basically, our best practices are for volunteers to get ready to map before the actual event so we can do a tutorial as a group and move forward together.  This new process does not allow for us to actually follow our own best practices which is really frustrating for both facilitators and mappers.  I've gotten like 5-6 angry emails about this just this week and a call a day so whatever you guys can do to make this a smoother process would be much appreciated.  Thanks!
Thanks! - Rachel

**Comment by xamanu** [comment]
Thanks @rachblevine for the feedback!

1) Could you possible make it open in a new tab / do a pop up window? People are being taken away from their exact task and it's adding to the confusion.

Sure, we could introduce this as a short fix. Just wondering whether this might be confusing to others, as after the tab with the email has been closed, people would have to reload the first page.

2) I don't think confirmation is necessary. Either you trust them or you don't, but making them wait for another confirmation email is delaying events and really putting people off right at the start of their engagement with this tool and HOT.

We took it away. People can map without email verification for now. They can also put a wrong email in there. 
Although I do think we need to verify emails! And we should implement measurements for this. It is not a matter of trust, it is a matter of verifying that there is no typo in the email and that people actually receive the messages.

3) Don't put the notification in a secondary tab (it's currently under the map tab). That means a new volunteer needs to start the mapping process to find this new requirement and pause. This make it complicated to do ahead of time.

The redesign will address this.

4) As it currently stands, the TM main page & individual task pages don't let you know anything else is needed. 

Right next to the username there should be a sign, that indicates that something is wrong and people should visit their profile to add their email address. With the redesign, it will be a pop-up right after login.

**Comment by xamanu** [comment]
Hi @xivk,

> Is this an absolute requirement to give up you email? Wouldn't it be possible to just encourage users to give their email or have an opt-out? Most people understand the need for email address and notifications coming directly from the tasking manager but it would be nice to have an opt-out.

In the tech team and together with the user experience analysts we consider it essential for the TM to have a way to talk to mappers, give them feedback on their mapping and allowing the validator or project manager to ask questions about work done. We see this as an important ingredient for new mappers to learn and also improve on their work and data they produce - a core functionality of the TM. Yes, people should be able to opt-out of all "not that important" messages. But the tool implies that questions can be asked to the mappers in the validation process, and it is essential to the behaviour that those messages reach the people to achieve the interaction the TM is built for.

**Comment by xamanu** [comment]
1. Yes, this is why I think the current compromise seems to be a good solution.
2. Just checked again. It is possible to map without having verified the email address. Is this just me? Can please check anybody else?
3. We are working on it to have a testing version ready for end of September. And then it may take some weeks to finish it up, write documentation, translate them and test it thoroughly.

**Comment by xamanu** [comment]
We've reviewed this extensively and tested over several rounds in #1975. We received very positive feedback. Closing here, as this is going to be part of TM4.
